src/storage/txn/commands/check_txn_status.rs

```diff
@@ -58,6 +58,9 @@ impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for CheckTxnStatus {
     /// the `current_ts` is literally the timestamp when this function is invoked. It may not be
     /// accurate.
     fn process_write(mut self, snapshot: S, context: WriteContext<'_, L>) -> Result<WriteResult> {
+        let new_max_ts = cmp::max(self.lock_ts, self.current_ts);
```

Comment:
What if `current_ts == max_uint64 > caller_start_ts > lock_ts`?

Reply:
Ah... I forgot actually both `current_ts` and `caller_start_ts` can be max uint64.